src/content_script/mastodonInject.js

```diff
@@ -129,20 +129,26 @@ async function injectInteractionButtons() {
 async function init() {
     injectFollowButton();
 
-    const observer = new MutationObserver(() => {
-        Promise.allSettled([
-            injectInteractionButtons(),
-        ]);
-    });
+    const ogType = document.querySelector("meta[property='og:type']");
 
-    const feedElement = await waitForElement(
-        "#mastodon .item-list[role='feed']",
-        false,
-        TIMEOUT_DURATION
-    );
-    observer.observe(feedElement, {
-        childList: true, subtree: true,
-    });
+    if (ogType && ogType.getAttribute("content") === "article"){
+        injectInteractionButtons();
+    } else {
```

Comment:
```suggestion
        // otherwise listen to the feed for new posts
        } else {
```

Proper way (see "clean code") would be to refactor that into two methods, but let's fix the code first…